diff --git a/drivers/spi/spi.c b/drivers/spi/spi.c
index 72436ad41..f3b540259 100644
--- a/drivers/spi/spi.c
+++ b/drivers/spi/spi.c
@@ -3236,39 +3236,27 @@ static int __spi_validate(struct spi_device *spi, struct spi_message *message)
 		if (xfer->tx_buf) {
 			if (xfer->tx_nbits != SPI_NBITS_SINGLE &&
 				xfer->tx_nbits != SPI_NBITS_DUAL &&
-				xfer->tx_nbits != SPI_NBITS_QUAD){
-				dev_warn(&spi->dev, "tx_nbits incorrect value");
-				xfer->tx_nbits = SPI_NBITS_SINGLE;
-			}
+				xfer->tx_nbits != SPI_NBITS_QUAD)
+				return -EINVAL;
 			if ((xfer->tx_nbits == SPI_NBITS_DUAL) &&
-				!(spi->mode & (SPI_TX_DUAL | SPI_TX_QUAD))){
-				dev_warn(&spi->dev, "tx_nbits requests incorrect mode (DUAL)");
-				xfer->tx_nbits = SPI_NBITS_SINGLE;
-			}
+				!(spi->mode & (SPI_TX_DUAL | SPI_TX_QUAD)))
+				return -EINVAL;
 			if ((xfer->tx_nbits == SPI_NBITS_QUAD) &&
-				!(spi->mode & SPI_TX_QUAD)){
-				dev_warn(&spi->dev, "tx_nbits requests incorrect mode (QUAD)");
-				xfer->tx_nbits = SPI_NBITS_SINGLE;
-			}
+				!(spi->mode & SPI_TX_QUAD))
+				return -EINVAL;
 		}
 		/* check transfer rx_nbits */
 		if (xfer->rx_buf) {
 			if (xfer->rx_nbits != SPI_NBITS_SINGLE &&
 				xfer->rx_nbits != SPI_NBITS_DUAL &&
-				xfer->rx_nbits != SPI_NBITS_QUAD){
-				dev_warn(&spi->dev, "rx_nbits incorrect value");
-				xfer->rx_nbits = SPI_NBITS_SINGLE;
-			}
+				xfer->rx_nbits != SPI_NBITS_QUAD)
+				return -EINVAL;
 			if ((xfer->rx_nbits == SPI_NBITS_DUAL) &&
-				!(spi->mode & (SPI_RX_DUAL | SPI_RX_QUAD))){
-				dev_warn(&spi->dev, "rx_nbits requests incorrect mode (DUAL)");
-				xfer->rx_nbits = SPI_NBITS_SINGLE;
-			}
+				!(spi->mode & (SPI_RX_DUAL | SPI_RX_QUAD)))
+				return -EINVAL;
 			if ((xfer->rx_nbits == SPI_NBITS_QUAD) &&
-				!(spi->mode & SPI_RX_QUAD)){
-				dev_warn(&spi->dev, "rx_nbits requests incorrect mode (QUAD)");
-				xfer->rx_nbits = SPI_NBITS_SINGLE;
-			}
+				!(spi->mode & SPI_RX_QUAD))
+				return -EINVAL;
 		}
 
 		if (xfer->word_delay_usecs < spi->word_delay_usecs)
